receiver/jaegerreceiver/trace_receiver_test.go

```diff
@@ -68,68 +73,74 @@ func TestReception(t *testing.T) {
-				Message: "Stale indices",
-			},
-			Links: []trace.Link{
-				{
-					TraceID: trace.TraceID{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80},
-					SpanID:  trace.SpanID{0xCF, 0xCE, 0xCD, 0xCC, 0xCB, 0xCA, 0xC9, 0xC8},
-					Type:    trace.LinkTypeParent,
-				},
-			},
-		},
-		{
-			SpanContext: trace.SpanContext{
-				TraceID: trace.TraceID{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80},
-				SpanID:  trace.SpanID{0xCF, 0xCE, 0xCD, 0xCC, 0xCB, 0xCA, 0xC9, 0xC8},
-			},
-			Name:      "ProxyFetch",
-			StartTime: nowPlus10min,
-			EndTime:   nowPlus10min2sec,
-			Status: trace.Status{
-				Code:    trace.StatusCodeInternal,
-				Message: "Frontend crash",
-			},
-			Links: []trace.Link{
-				{
-					TraceID: trace.TraceID{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80},
-					SpanID:  trace.SpanID{0xAF, 0xAE, 0xAD, 0xAC, 0xAB, 0xAA, 0xA9, 0xA8},
-					Type:    trace.LinkTypeChild,
-				},
-			},
-		},
-	}
-
-	for _, sd := range spandata {
+	for _, sd := range traceFixture(now, nowPlus10min, nowPlus10min2sec) {
 		jexp.ExportSpan(sd)
 	}
 	jexp.Flush()
 
-	// Simulate and account for network latency but also the reception process on the server.
-	<-time.After(500 * time.Millisecond)
+	got := sink.AllTraces()
+	want := expectedTraceData(now, nowPlus10min, nowPlus10min2sec)
 
-	jexp.Flush()
+	if diff := cmp.Diff(got, want); diff != "" {
+		t.Errorf("Mismatched responses\n-Got +Want:\n\t%s", diff)
+	}
+}
+
+func TestGRPCReception(t *testing.T) {
+	// prepare
+	config := &Configuration{
+		CollectorGRPCPort: 14250, // that's the only one used by this test
+	}
+	sink := new(exportertest.SinkTraceExporter)
+
+	jr, err := New(context.Background(), config, sink)
+	assert.NoError(t, err, "should not have failed to create a new receiver")
+	defer jr.StopTraceReception()
+
+	mh := receivertest.NewMockHost()
+	err = jr.StartTraceReception(mh)
+	assert.NoError(t, err, "should not have failed to start trace reception")
+	t.Log("StartTraceReception")
+
+	conn, err := grpc.Dial(fmt.Sprintf("0.0.0.0:%d", config.CollectorGRPCPort), grpc.WithInsecure())
+	defer conn.Close()
+
+	cl := api_v2.NewCollectorServiceClient(conn)
+
+	now := time.Unix(1542158650, 536343000).UTC()
+	d10min := 10 * time.Minute
+	d2sec := 2 * time.Second
+	nowPlus10min := now.Add(d10min)
+	nowPlus10min2sec := now.Add(d10min).Add(d2sec)
+
+	// test
+	req := grpcFixture(now, d10min, d2sec)
+	resp, err := cl.PostSpans(context.Background(), req, grpc.WaitForReady(true))
+
+	// verify
+	assert.NoError(t, err, "should not have failed to post spans")
+	assert.NotNil(t, resp, "response should not have been nil")
 
 	got := sink.AllTraces()
+	want := expectedTraceData(now, nowPlus10min, nowPlus10min2sec)
+
+	assert.Len(t, req.Batch.Spans, len(want[0].Spans), "got a conflicting amount of spans")
+
+	if diff := cmp.Diff(got, want); diff != "" {
```

Comment:
There is no way for this "expected" OC proto to have come from Jaeger, so it is fine to remove it from the expected data. That said the question about what to do when converting from OC to Jaeger remains: in principle it seems that at most it could be represented via some attribute/tags. Let's open an issue separately from this PR to decide the approach on that case.

Reply:
Created #206